docs/best-practices/gitops-with-kruise.md

```diff
@@ -173,13 +118,143 @@ data:
+  - name: argocd-secret
+    secret:
+      secretName: argosecret
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: helloworld-pipeline
+spec:
+  params:
+  - name: gitrepositoryurl
+    type: string
+  - name: branch
+    type: string
+  - name: short_sha
+    type: string
+  - name: docker_repo
+    type: string
+  - name: app_name
+    type: string
+  - name: app_ns
+    type: string
+  - name: k8s_server
+    type: string
+  # Here you can connect with CI process to realize CI/CD Pipeline
+  tasks:
+  - name: helloworld-argocd
+    taskRef:
+      name: helloworld-argocd
+    params:
+    - name: gitrepositoryurl
+      value: $(params.gitrepositoryurl)
+    - name: short_sha
+      value: $(params.short_sha)
+    - name: branch
+      value: $(params.branch)
+    - name: docker_repo
+      value: $(params.docker_repo)
+    - name: app_name
+      value: $(params.app_name)
+    - name: app_ns
+      value: $(params.app_ns)
+    - name: k8s_server
+      value: $(params.k8s_server)
+```
+### Run Tekton Pipeline
+Configure PipelineRun CRD, and kubectl apply -f in k8s cluster to run Pipeline, as follows:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: helloworld-pipeline-run-1
+spec:
+  pipelineRef:
+    name: helloworld-pipeline
+  params:
+  - name: gitrepositoryurl
+    value: https://github.com/zmberg/samples.git
+  - name: branch
+    value: hello_world
+  - name: short_sha
+    value: d92ae174b
+  - name: docker_repo
+    value: zhaomingshan/kruise
+  - name: app_name
+    value: helloworld
+  - name: app_ns
+    value: helloworld
+  - name: k8s_server
+    value: https://kubernetes.default.svc
+```
+The results can be viewed via the argo-cd cli, as follows:
 
 ![guestbook](/img/docs/best-practices/argo_sync_healthy.png)
 
 ## Summary
 OpenKruise is more of a Kubernetes level extended capability, such as in-place upgrade, preImageDownload, etc. So many community users using OpenKruise in production environments have some additional costs, need to integrate or self-research container PaaS.
 The main purpose of this article is to combine some of the best Paas solutions in the community with OpenKruise, so that as many people as possible get to enjoy the dividends of cloud-native at a lower cost.
 
-Argo-cd is a very good product for the community, and it is very user-friendly for OpenKruise's many custom CRD resources. This article is hope that many partners in the community can provide more ideas in "How to use OpenKruise easily", and also we will try to combine it with some CI/CD pipelines later to better practice the devops concept.
+Argo-cd is a very good product for the community, and it is very user-friendly for OpenKruise's many custom CRD resources. This article is hope that many partners in the community can provide more ideas in "How to use OpenKruise easily", and also we will try to combine it with some other CI/CD pipelines later to better practice the devops concept.
```

Comment:
> it is very user-friendly for OpenKruise's many custom CRD resources. This article is hope that many partners in the community can provide more ideas in "How to use OpenKruise easily", and also we will try to combine it with some other CI/CD pipelines later to better practice the devops concept.

Argo-CD integrate well with OpenKruise's CRD resources, we hope that this article can intrigue more ideas from the community about  "How to use OpenKruise easily". We will try to integrate OpenKruise with other CI/CD pipelines later to better implement DevOps practice.